snappy/hwaccess_test.go

```diff
@@ -287,3 +287,29 @@ func (s *SnapTestSuite) TestRemoveAllHWAccess(c *C) {
 	c.Check(helpers.FileExists(filepath.Join(dirs.SnapAppArmorAdditionalDir, "hello-app.hwaccess.yaml")), Equals, false)
 	c.Check(*regenerateAppArmorRulesWasCalled, Equals, true)
 }
+
+func (s *SnapTestSuite) TestAddSysDevice(c *C) {
+	makeInstalledMockSnap(s.tempdir, "")
+	regenerateAppArmorRulesWasCalled := mockRegenerateAppArmorRules()
+
+	err := AddHWAccess("hello-app", "/sys/devices/foo1")
+	c.Assert(err, IsNil)
+	err = AddHWAccess("hello-app", "/sys/class/gpio/foo2")
+	c.Assert(err, Equals, nil)
```

Comment:
Sorry, just a typo from the negative test I've done before committing. I will fix it